<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.38.2" />
  <meta name="author" content="Computational Biology Lab">

  
  
  
  
    
      
    
  
  <meta name="description" content="Table of Contents  Introduction 1. Hypergeometric test and gene set enrichment analysis  1.1 Prepare the input data 1.2 Initialize and preprocess 1.3 Perform analysis  2. Enriched subnetwork analysis  2.1 Prepare input, initialize and preprocess 3.2 Perform analysis    Introduction This case study uses a time series CRISPR genome-wide drop-out data as a demonstration to perform time series analysis. Data ‘d7’, ‘d13’ and ‘d25’ are three gRNA sequencing data after transducting the CRISPR system into a human cancer cellline on day 7, day 13 and day 25 separately (Tzelepis K (2016)), they are further preprocessed by MAGeCK to identify significant essential genes from genome-scale CRISPR knockout screens.">

  
  <link rel="alternate" hreflang="en-us" href="/tutorials/casestudy2/">

  


  

  
  
  <meta name="theme-color" content="#4caf50">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.1/css/academicons.min.css" integrity="sha512-NThgw3XKQ1absAahW6to7Ey42uycrVvfNfyjqcFNgCmOCQ5AR4AO0SiXrN+8ZtYeappp56lk1WtvjVmEa+VR6A==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7cMerriweather%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  

  
  <link rel="alternate" href="/index.xml" type="application/rss+xml" title="HTSanalyzeR2">
  <link rel="feed" href="/index.xml" type="application/rss+xml" title="HTSanalyzeR2">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/tutorials/casestudy2/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="og:site_name" content="HTSanalyzeR2">
  <meta property="og:url" content="/tutorials/casestudy2/">
  <meta property="og:title" content="Case Study 2: Time series analysis for CRISPR data | HTSanalyzeR2">
  <meta property="og:description" content="Table of Contents  Introduction 1. Hypergeometric test and gene set enrichment analysis  1.1 Prepare the input data 1.2 Initialize and preprocess 1.3 Perform analysis  2. Enriched subnetwork analysis  2.1 Prepare input, initialize and preprocess 3.2 Perform analysis    Introduction This case study uses a time series CRISPR genome-wide drop-out data as a demonstration to perform time series analysis. Data ‘d7’, ‘d13’ and ‘d25’ are three gRNA sequencing data after transducting the CRISPR system into a human cancer cellline on day 7, day 13 and day 25 separately (Tzelepis K (2016)), they are further preprocessed by MAGeCK to identify significant essential genes from genome-scale CRISPR knockout screens."><meta property="og:image" content="/img/tutorials/case2.jpg">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2016-04-20T00:00:00&#43;08:00">
  
  <meta property="article:modified_time" content="2018-04-11T00:00:00&#43;08:00">
  

  

  <title>Case Study 2: Time series analysis for CRISPR data | HTSanalyzeR2</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">HTSanalyzeR2</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#R_language">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Tutorials</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/tutorials/casestudy1">
                
                <span>Case Study1</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/tutorials/casestudy2">
                
                <span>Case Study2</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/tutorials/casestudy3">
                
                <span>Case Study3</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Resources</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/resources/prepro">
                
                <span>Preprocessing</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/resources/hyper">
                
                <span>Hypergeometrics</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/resources/es">
                
                <span>Enrichment Subnetwork</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Support</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Supported Analysis</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Data Type</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Supported Pathways</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Supported Species</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Visualization</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Term of service</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/service/session">
                
                <span>Session info</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/service/ref">
                
                <span>References</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Contact us</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  
<div class="article-header">
  
  
    <img src="/img/tutorials/case2.jpg" class="article-banner" itemprop="image">
  

  <span class="article-header-caption"><a href="https://pro-essay-writer.com/blog/case-study" target="_blank">Image credit</a></span>
</div>



  <div class="article-container">
    <h1 itemprop="name">Case Study 2: Time series analysis for CRISPR data</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
        Last updated on
    
    <time datetime="2016-04-20 00:00:00 &#43;0800 CST" itemprop="datePublished dateModified">
      Apr 11, 2018
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="Computational Biology Lab">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    6 min read
  </span>
  

  
  

  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=Case%20Study%202%3a%20Time%20series%20analysis%20for%20CRISPR%20data&amp;url=%2ftutorials%2fcasestudy2%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2ftutorials%2fcasestudy2%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2ftutorials%2fcasestudy2%2f&amp;title=Case%20Study%202%3a%20Time%20series%20analysis%20for%20CRISPR%20data"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2ftutorials%2fcasestudy2%2f&amp;title=Case%20Study%202%3a%20Time%20series%20analysis%20for%20CRISPR%20data"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=Case%20Study%202%3a%20Time%20series%20analysis%20for%20CRISPR%20data&amp;body=%2ftutorials%2fcasestudy2%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      

<hr />

<p><span id="top"></span></p>

<h2>Table of Contents</h2>
<nav id="TableOfContents">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#1-hypergeometric-test-and-gene-set-enrichment-analysis">1. Hypergeometric test and gene set enrichment analysis</a>
<ul>
<li><a href="#1-1-prepare-the-input-data">1.1 Prepare the input data</a></li>
<li><a href="#1-2-initialize-and-preprocess">1.2 Initialize and preprocess</a></li>
<li><a href="#1-3-perform-analysis">1.3 Perform analysis</a></li>
</ul></li>
<li><a href="#2-enriched-subnetwork-analysis">2. Enriched subnetwork analysis</a>
<ul>
<li><a href="#2-1-prepare-input-initialize-and-preprocess">2.1 Prepare input, initialize and preprocess</a></li>
<li><a href="#3-2-perform-analysis">3.2 Perform analysis</a></li>
</ul></li>
</ul>
</nav>


<h1 id="introduction">Introduction</h1>

<p><p align="justify">This case study uses a time series CRISPR genome-wide drop-out data as a demonstration
to perform time series analysis. Data ‘d7’, ‘d13’ and ‘d25’ are three gRNA sequencing data
after transducting the CRISPR system into a human cancer cellline on day 7, day 13 and
day 25 separately (Tzelepis K (2016)), they are further preprocessed by <a href="https://sourceforge.net/p/mageck/wiki/Home/" target="_blank">MAGeCK</a> to identify
significant essential genes from genome-scale CRISPR knockout screens. Again, here to simply
the compilation of this vignette, we skip the preprocessed steps and start from the results
gotten by MAGeCK.</p>

<p><a href="#top"><i class="fa fa-hand-o-up fa-1x "></i>Top</a></p>

<h1 id="1-hypergeometric-test-and-gene-set-enrichment-analysis">1. Hypergeometric test and gene set enrichment analysis</h1>

<h2 id="1-1-prepare-the-input-data">1.1 Prepare the input data</h2>

<p><p align="justify">To perform analysis for time series data, one must prepare the following inputs:
<p align="justify">1. A character matrix contains experiment information with each experiment in row and
descriptions in column. Specifically, it should at least contain two columns named as
‘ID’ and ‘Description’.
<p align="justify">2. A list of phenotypes, each element of this list is a phenotype of that experiment. An
important thing here needs to be noted is the order of each element of this list
must match the order of ‘ID’ in the experiment information matrix.
<p align="justify">3. A list of gene set collections which can either be gotten by HTSanalyzeR2 or defined
by users using customized gene sets.</p>

<p><p align="justify">To make it easy to compile this vignette, here we only use Biological Process (BP) in Gene
Ontology to make a demonstration.</p>

<pre><code>data(d7, d13, d25)
expInfor &lt;- matrix(c(&quot;d7&quot;, &quot;d13&quot;, &quot;d25&quot;), nrow = 3, ncol = 2, byrow = FALSE,
dimnames = list(NULL, c(&quot;ID&quot;, &quot;Description&quot;)))
datalist &lt;- list(d7, d13, d25)
phenotypeTS &lt;- lapply(datalist, function(x) {
tmp &lt;- as.vector(x$neg.lfc)
names(tmp) &lt;- x$id
tmp
})
GO_BP &lt;- GOGeneSets(species=&quot;Hs&quot;, ontologies=c(&quot;BP&quot;))
ListGSC &lt;- list(GO_BP=GO_BP)
</code></pre>

<p><p align="justify">Similar as single dataset analysis, if you also want to do hypergeometric test, a list of hits
is needed. Here, each element of this list is a hits of that experiment. Also, <strong>the order of each element of this list must match the order of ‘ID’ in the experiment information matrix.</strong> Here, for each data set, we define genes with pvalue less than 0.01 as hits.</p>

<pre><code>hitsTS &lt;- lapply(datalist, function(x){
tmp &lt;- x[x$neg.p.value &lt; 0.01, &quot;id&quot;]
tmp
})
</code></pre>

<p><a href="#top"><i class="fa fa-hand-o-up fa-1x "></i>Top</a></p>

<h2 id="1-2-initialize-and-preprocess">1.2 Initialize and preprocess</h2>

<p><p align="justify">To perform gene set enrichment analysis and hypermetric test for time-series data, an S4
class ‘GSCABatch’ which can pack the time series data to do further analysis is developed.
First, you need to create a new ‘GSCABatch’ object using the prepared inputs.</p>

<pre><code>gscaTS &lt;- GSCABatch(expInfor = expInfor,
phenotypeTS = phenotypeTS, listOfGeneSetCollections = ListGSC,
hitsTS = hitsTS)
</code></pre>

<p><p align="justify">Then, the ‘GSCABatch’ object need to be preprocessed using preprocessGscaTS method. The
preprocess procedure here is the same as single data set. This step would return a list of
preprocessed ‘GSCA’ object.</p>

<pre><code>gscaTS1 &lt;- preprocessGscaTS(gscaTS, species=&quot;Hs&quot;, initialIDs=&quot;SYMBOL&quot;,
keepMultipleMappings=TRUE,
duplicateRemoverMethod=&quot;max&quot;,
orderAbsValue=FALSE)
</code></pre>

<p><a href="#top"><i class="fa fa-hand-o-up fa-1x "></i>Top</a></p>

<h2 id="1-3-perform-analysis">1.3 Perform analysis</h2>

<p><p align="justify">After getting a list of preprocessed ‘GSCA’ object, you can use analyzeGscaTS to perform
hypergeometric test as well as GSEA on it. The parameters’ function here is the same as in
single data set. Similarly, to speed up you can use multiple cores via doParallel package. This
step would return a list of analyzed ‘GSCA’ object.</p>

<pre><code>## analyze using 4 cores
if (requireNamespace(&quot;doParallel&quot;, quietly=TRUE)) {
doParallel::registerDoParallel(cores=4)
} else {
}
gscaTS2 &lt;- analyzeGscaTS(gscaTS1, para=list(pValueCutoff=0.05,
pAdjustMethod=&quot;BH&quot;,
nPermutations=100,
minGeneSetSize=100,
exponent=1),
doGSOA = TRUE, doGSEA = TRUE)
## GSEA result of the first experiment usingBP gene sets
head(getResult(gscaTS2[[1]])$GSEA.results$GO_BP, 3)
</code></pre>

<p><p align="justify">To make the result more understandable, users are highly recommended to annotate the gene
sets ID to names by function appendGSTermsTS. As a result, an additional column named
‘Gene.Set.Term’ would appear.</p>

<pre><code>gscaTS3 &lt;- appendGSTermsTS(gscaTS2, goGSCs=c(&quot;GO_BP&quot;))
head(getResult(gscaTS3[[1]])$GSEA.results$GO_BP, 3)
## Gene.Set.Term Observed.score Pvalue
## GO:0006281 DNA repair -0.6898409 0
## GO:0001525 angiogenesis -0.1840823 0
## GO:0000398 mRNA splicing, via spliceosome -0.8693455 0
## Adjusted.Pvalue
## GO:0006281 0
## GO:0001525 0
## GO:0000398 0
</code></pre>

<p><p align="justify">You can then use reportAll to generate an interactive Shiny report to visualize a union
enrichment map for this time series data. To put it more specific, a union enrichment map is
generated by taking the union of significant gene sets in each experiment and then form an
enrichment map as illustrated before. Thus there maybe be some gene sets not significant in
one time point but in others. More details please see Part4: An interactive Shiny report of
results.</p>

<p><a href="#top"><i class="fa fa-hand-o-up fa-1x "></i>Top</a></p>

<h1 id="2-enriched-subnetwork-analysis">2. Enriched subnetwork analysis</h1>

<h2 id="2-1-prepare-input-initialize-and-preprocess">2.1 Prepare input, initialize and preprocess</h2>

<p><p align="justify">An S4 class named ‘NWABatch’ is developed to pack time series data for further subnetwork
analysis. You need first to create a new object of class ‘NWABatch’. To this end, a list of
pvalues is needed. Each element of this list is a vector of pvalues of that experiment. <strong>Again, an important thing needs to be noted is the order of each element of this list must match the order of ‘ID’ in the experiment information matrix.</strong> If a list of phenotypes
is also available, they can be inputted during the initialization stage and used to highlight
nodes with different colors in the identified subnetwork. Also, the order of each element of
this phenotypes list must match the order of ‘ID’ in the experiment information matrix.</p>

<pre><code>pvalueTS &lt;- lapply(datalist, function(x){
tmp &lt;- as.vector(x$neg.p.value)
names(tmp) &lt;- x$id
tmp
})
## generate a new 'NWABatch' object for further analysis
nwaTS &lt;- NWABatch(expInfor = expInfor,
pvalueTS = pvalueTS, phenotypeTS = phenotypeTS)
</code></pre>

<p><p align="justify">After creating an object of ‘NWABatch’, a preprocessing step needs to be performed which
will return a list of preprocessed ‘NWA’ objects.</p>

<pre><code>nwaTS1 &lt;- preprocessNwaTS(nwaTS, species=&quot;Hs&quot;, initialIDs=&quot;SYMBOL&quot;,
keepMultipleMappings=TRUE,
duplicateRemoverMethod=&quot;max&quot;)
</code></pre>

<p><a href="#top"><i class="fa fa-hand-o-up fa-1x "></i>Top</a></p>

<h2 id="3-2-perform-analysis">3.2 Perform analysis</h2>

<p><p align="justify">Similarly, an interactome needs to be created before performing subnetwork analysis using
<em>interactomeNwaTS</em> if you have not inputted your own interactome in the initial step. You can
either specify the species and fetch the corresponding network from BioGRID database, or input
an interaction matrix if it is in right format. More details please see <em>help(interactomeNwaTS)</em>.</p>

<p><p align="justify">Then, <em>analyzeNwaTS</em> could perform the subnetwork analysis for a list of ‘NWA’ object, which
would take a few minutes. Finally, this step would return a list of analyzed ‘NWA’ objects.</p>

<pre><code>nwaTS2 &lt;- interactomeNwaTS(nwaTS1, species=&quot;Hs&quot;,
reportDir=&quot;HTSanalyzerReport&quot;, genetic=FALSE)
nwaTS3 &lt;- analyzeNwaTS(nwaTS2, fdr=0.0001, species=&quot;Hs&quot;)
</code></pre>

<pre><code>## get a general summary for the first experiment
summarize(nwaTS3[[1]])
</code></pre>

<p><p align="justify">You can then use reportAll to generate an interactive Shiny report to visualize a union
subnetwork for this time series data. To put it more specific, a union subnetwork is generated
by taking the union of identified subnetwork in each experiment. Thus there maybe be some
genes not identified in the subnetwork of one time point but in others. More details please
see Part4: An interactive Shiny report of results.{to be modified}</p>

<p><a href="#top"><i class="fa fa-hand-o-up fa-1x "></i>Top</a></p>

<hr />

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/tags/casestudy" target="_blank">caseStudy</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/tutorials/casestudy1/">Case Study 1: Single dataset analysis for gene expression data</a></li>
        
        <li><a href="/tutorials/casestudy3/">Special Usage of HTSanalyzeR2</a></li>
        
      </ul>
    </div>
    

    

    


  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Copyright© 2018 Lin QI of Dr.Xin Wang's Lab. zzx. All rights reserved. 

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    
    <script async defer src="//maps.googleapis.com/maps/api/js?key=AIzaSyD3AGpKDAfo_LjBRHd-YkYmMjBDlIHfdlo"></script>
    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gmaps.js/0.4.25/gmaps.min.js" integrity="sha256-7vjlAeb8OaTrCXZkCNun9djzuB2owUsaO72kXaFDBJs=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    

  </body>
</html>

