

<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.38.2" />
  <meta name="author" content="Computational Biology Lab">

  
  
  
  
  <meta name="description" content="">

  
  <link rel="alternate" hreflang="en-us" href="/service/">

  


  

  
  
  <meta name="theme-color" content="#4caf50">
  
  
  
  
    
  
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.1/css/academicons.min.css" integrity="sha512-NThgw3XKQ1absAahW6to7Ey42uycrVvfNfyjqcFNgCmOCQ5AR4AO0SiXrN+8ZtYeappp56lk1WtvjVmEa+VR6A==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7cMerriweather%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  

  
  <link rel="alternate" href="/service/index.xml" type="application/rss+xml" title="HTSanalyzeR2">
  <link rel="feed" href="/service/index.xml" type="application/rss+xml" title="HTSanalyzeR2">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/service/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="og:site_name" content="HTSanalyzeR2">
  <meta property="og:url" content="/service/">
  <meta property="og:title" content="Service | HTSanalyzeR2">
  <meta property="og:description" content="">
  <meta property="og:locale" content="en-us">
  
  <meta property="og:updated_time" content="2018-04-11T00:00:00&#43;08:00">
  

  

  <title>Service | HTSanalyzeR2</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">HTSanalyzeR2</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#hero">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Tutorials</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/tutorials/casestudy1">
                
                <span>Case Study1</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/tutorials/casestudy2">
                
                <span>Case Study2</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/tutorials/casestudy3">
                
                <span>Case Study3</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Resources</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/resources/prepro">
                
                <span>Preprocessing</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/resources/hyper">
                
                <span>Hypergeometrics</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/resources/es">
                
                <span>Enrichment Subnetwork</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Support</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Supported Analysis</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Data Type</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Supported Pathways</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Supported Species</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/support/">
                
                <span>Visualization</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Term of service</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/service/session">
                
                <span>Session info</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/service/ref">
                
                <span>References</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Contact us</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>





<div class="universal-wrapper">

  <h1>Service</h1>

  
  <div class="article-style" itemprop="articleBody"><hr />

<div class="alert alert-note">
  <p>[<strong>Session info:</strong>]</p>

</div>


<blockquote>
<p>## R version 3.5.1 (2018-07-02)<br>
## Platform: x86_64-pc-linux-gnu (64-bit)<br>
## Running under: Ubuntu 18.04 LTS<br>
##
## Matrix products: default<br>
## BLAS: /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3<br>
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so<br>
##<br>
## locale:<br>
## [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C<br>
## [3] LC_TIME=en_US.UTF-8 LC_COLLATE=en_US.UTF-8<br>
## [5] LC_MONETARY=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8<br>
## [7] LC_PAPER=en_US.UTF-8 LC_NAME=C<br>
## [9] LC_ADDRESS=C LC_TELEPHONE=C<br>
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C<br>
##<br>
## attached base packages:<br>
## [1] parallel stats4 stats graphics grDevices utils datasets<br>
## [8] methods base<br>
##<br>
## other attached packages:<br>
## [1] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2<br>
## [2] GenomicFeatures_1.32.0<br>
## [3] GenomicRanges_1.32.3<br>
## [4] GenomeInfoDb_1.16.0<br>
## [5] igraph_1.2.1<br>
## [6] GO.db_3.6.0<br>
## [7] KEGGREST_1.20.0<br>
## [8] org.Hs.eg.db_3.6.0<br>
## [9] AnnotationDbi_1.42.1<br>
## [10] IRanges_2.14.10<br>
## [11] S4Vectors_0.18.3<br>
## [12] Biobase_2.40.0<br>
## [13] BiocGenerics_0.26.0<br>
## [14] HTSanalyzeR2_0.99.15<br>
## [15] BiocStyle_2.8.2<br>
##<br>
## loaded via a namespace (and not attached):<br>
## [1] fgsea_1.6.0 colorspace_1.3-2<br>
## [3] hwriter_1.3.2 rprojroot_1.3-2<br>
## [5] XVector_0.20.0 affyio_1.50.0<br>
## [7] DT_0.4 bit64_0.9-7<br>
## [9] mvtnorm_1.0-8 codetools_0.2-15<br>
## [11] splines_3.5.1 doParallel_1.0.11<br>
## [13] robustbase_0.93-1.1 knitr_1.20<br>
## [15] splots_1.46.0 Rsamtools_1.32.0<br>
## [17] prada_1.56.0 annotate_1.58.0<br>
## [19] cluster_2.0.7-1 Rmpfr_0.7-0<br>
## [21] vsn_3.48.1 png_0.1-7<br>
## [23] shinydashboard_0.7.0 graph_1.58.0<br>
## [25] shiny_1.1.0 rrcov_1.4-4<br>
## [27] compiler_3.5.1 httr_1.3.1<br>
## [29] backports_1.1.2 assertthat_0.2.0<br>
## [31] Matrix_1.2-14 lazyeval_0.2.1<br>
## [33] limma_3.36.2 later_0.7.3<br>
## [35] prettyunits_1.0.2 htmltools_0.3.6<br>
## [37] tools_3.5.1 gmp_0.5-13.2<br>
## [39] bindrcpp_0.2.2 GenomeInfoDbData_1.1.0<br>
## [41] gtable_0.2.0 glue_1.3.0<br>
## [43] affy_1.58.0 Category_2.46.0<br>
## [45] dplyr_0.7.6 fastmatch_1.1-0<br>
## [47] Rcpp_0.12.17 Biostrings_2.48.0<br>
## [49] preprocessCore_1.42.0 rtracklayer_1.40.3<br>
## [51] iterators_1.0.10 xfun_0.3<br>
## [53] stringr_1.3.1 mime_0.5<br>
## [55] miniUI_0.1.1.1 XML_3.98-1.12<br>
## [57] BioNet_1.40.0 DEoptimR_1.0-8<br>
## [59] zlibbioc_1.26.0 MASS_7.3-50<br>
## [61] scales_0.5.0 RankProd_3.6.0<br>
## [63] BiocInstaller_1.30.0 colourpicker_1.0<br>
## [65] hms_0.4.2 promises_1.0.1<br>
## [67] SummarizedExperiment_1.10.1 RBGL_1.56.0<br>
## [69] RColorBrewer_1.1-2 curl_3.2<br>
## [71] yaml_2.1.19 memoise_1.1.0<br>
## [73] gridExtra_2.3 ggplot2_3.0.0<br>
## [75] biomaRt_2.36.1 stringi_1.2.3<br>
## [77] RSQLite_2.1.1 genefilter_1.62.0<br>
## [79] cellHTS2_2.44.0 pcaPP_1.9-73<br>
## [81] foreach_1.4.4 BiocParallel_1.14.1<br>
## [83] matrixStats_0.53.1 rlang_0.2.1<br>
## [85] pkgconfig_2.0.1 bitops_1.0-6<br>
## [87] evaluate_0.11 lattice_0.20-35<br>
## [89] purrr_0.2.5 bindr_0.1.1<br>
## [91] GenomicAlignments_1.16.0 htmlwidgets_1.2<br>
## [93] bit_1.1-14 tidyselect_0.2.4<br>
## [95] GSEABase_1.42.0 plyr_1.8.4<br>
## [97] magrittr_1.5 bookdown_0.7<br>
## [99] R6_2.2.2 DelayedArray_0.6.1<br>
## [101] DBI_1.0.0 pillar_1.3.0<br>
## [103] survival_2.42-3 RCurl_1.95-4.11<br>
## [105] tibble_1.4.2 crayon_1.3.4<br>
## [107] rmarkdown_1.10 progress_1.2.0<br>
## [109] locfit_1.5-9.1 grid_3.5.1<br>
## [111] data.table_1.11.4 blob_1.1.1<br>
## [113] digest_0.6.15 xtable_1.8-2<br>
## [115] httpuv_1.4.5 munsell_0.5.0<br></p>
</blockquote>

<hr />

<div class="alert alert-note">
  <p>[<strong>References:</strong>]</p>

</div>


<p>Arthur Liberzon, Reid Pinchback, Aravind Subramanian. 2011. “Molecular Signatures
Database (Msigdb) 3.0.” Bioinformatics 27 (12): 1739–40. doi:<a href="https://doi.org/10.1093/bioinformatics/btr260" target="_blank">10.1093/bioinformatics/btr260</a>.<br></p>

<p>Beisser, Klau, D. 2010. “BioNet: An R-Package for the Functional Analysis of Biological
Networks.” Bioinformatics 26 (8): 1129–30. doi:<a href="https://doi.org/10.1093/bioinformatics/btq089" target="_blank">10.1093/bioinformatics/btq089</a>.<br></p>

<p>Dittrich MT, Rosenwald A, Klau GW. 2008. “Identifying Functional Modules in Protein–
Protein Interaction Networks: An Integrated Exact Approach.” Bioinformatics 24 (13):
i223–i231. doi:<a href="https://academic.oup.com/bioinformatics/article/24/13/i223/231653" target="_blank">10.1093/bioinformatics/btn161</a>.<br></p>

<p>Guinney J, Wang X, Dienstmann R. 2015. “The Consensus Molecular Subtypes of Colorectal
Cancer.” Nature Medicine 21 (11): 1350–6. doi:<a href="https://www.nature.com/articles/nm.3967" target="_blank">10.1038/nm.3967</a>.<br></p>

<p>Merico D, Stueker O, Isserlin R. 2010. “Enrichment Map: A Network-Based Method
for Gene-Set Enrichment Visualization and Interpretation.” PLoS ONE 5 (11): e13984.
doi:<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0013984" target="_blank">10.1371/journal.pone.0013984</a>.<br></p>

<p>Sergushichev, Alexey. 2016. “An Algorithm for Fast Preranked Gene Set Enrichment Analysis
Using Cumulative Statistic Calculation.” bioRxiv. doi:<a href="https://doi.org/10.1101/060012" target="_blank">10.1101.060012</a>.<br></p>

<p>Subramanian A, Mootha VK, Tamayo P. 2005. “Gene Set Enrichment Analysis: A
Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.” Proceedings
of the National Academy of Sciences of the United States of America 102 (43): 15545–50.
doi:<a href="https://doi.org/10.1073/pnas.0506580102" target="_blank">10.1073/pnas.0506580102</a>. <br></p>

<p>Tzelepis K, De Braekeleer E, Koike-Yusa H. 2016. “A Crispr Dropout Screen Identifies Genetic
Vulnerabilities and Therapeutic Targets in Acute Myeloid Leukemia.” Cell Reports 17 (4):
1193–1205. doi:<a href="https://doi.org/10.1016/j.celrep.2016.09.079" target="_blank">10.1016/j.celrep.2016.09.079</a>.<br></p>
</div>
  

  
  
  <div>
    <h2><a href="/service/ref/">References</a></h2>
    <div class="article-style">
      
      [References:]
 Arthur Liberzon, Reid Pinchback, Aravind Subramanian. 2011. “Molecular Signatures Database (Msigdb) 3.0.” Bioinformatics 27 (12): 1739–40. doi:10.1093/bioinformatics/btr260.
Beisser, Klau, D. 2010. “BioNet: An R-Package for the Functional Analysis of Biological Networks.” Bioinformatics 26 (8): 1129–30. doi:10.1093/bioinformatics/btq089.
Dittrich MT, Rosenwald A, Klau GW. 2008. “Identifying Functional Modules in Protein– Protein Interaction Networks: An Integrated Exact Approach.” Bioinformatics 24 (13): i223–i231. doi:10.1093/bioinformatics/btn161.
Guinney J, Wang X, Dienstmann R. 2015. “The Consensus Molecular Subtypes of Colorectal Cancer.
      
    </div>
  </div>
  
  <div>
    <h2><a href="/service/session/">Session info</a></h2>
    <div class="article-style">
      
      <p>[Session info:]
  ## R version 3.5.1 (2018-07-02)</p>

<h2 id="platform-x86-64-pc-linux-gnu-64-bit">Platform: x86_64-pc-linux-gnu (64-bit)</h2>

<h2 id="running-under-ubuntu-18-04-lts">Running under: Ubuntu 18.04 LTS</h2>

<h2 id="matrix-products-default">## Matrix products: default</h2>

<h2 id="blas-usr-lib-x86-64-linux-gnu-openblas-libblas-so-3">BLAS: /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3</h2>

<h2 id="lapack-usr-lib-x86-64-linux-gnu-libopenblasp-r0-2-20-so">LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so</h2>

<p>##</p>

<h2 id="locale">locale:</h2>

<h2 id="1-lc-ctype-en-us-utf-8-lc-numeric-c">[1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C</h2>

<h2 id="3-lc-time-en-us-utf-8-lc-collate-en-us-utf-8">[3] LC_TIME=en_US.UTF-8 LC_COLLATE=en_US.UTF-8</h2>

<h2 id="5-lc-monetary-en-us-utf-8-lc-messages-en-us-utf-8">[5] LC_MONETARY=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8</h2>

<h2 id="7-lc-paper-en-us-utf-8-lc-name-c">[7] LC_PAPER=en_US.UTF-8 LC_NAME=C</h2>

<h2 id="9-lc-address-c-lc-telephone-c">[9] LC_ADDRESS=C LC_TELEPHONE=C</h2>

<h2 id="11-lc-measurement-en-us-utf-8-lc-identification-c">[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C</h2>

<p>##</p>

<h2 id="attached-base-packages">attached base packages:</h2>

<h2 id="1-parallel-stats4-stats-graphics-grdevices-utils-datasets">[1] parallel stats4 stats graphics grDevices utils datasets</h2>

      
    </div>
  </div>
  

  


</div>
<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Copyright© 2018 Lin QI of Dr.Xin Wang's Lab. zzx. All rights reserved. 

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    
    <script async defer src="//maps.googleapis.com/maps/api/js?key=AIzaSyD3AGpKDAfo_LjBRHd-YkYmMjBDlIHfdlo"></script>
    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gmaps.js/0.4.25/gmaps.min.js" integrity="sha256-7vjlAeb8OaTrCXZkCNun9djzuB2owUsaO72kXaFDBJs=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    

    
    

  </body>
</html>



